src/magento/index.js

```diff
@@ -118,30 +85,38 @@ class Magento {
       fetch(uri, { method, headers, body: JSON.stringify(data) })
         .then(response => {
           console.log(response);
-          return response.json();
+          if (response.ok) {
+            return response.json();
+          }
+          // Possible 401 or other network error
+          return response.json().then(errorResponse => Promise.reject(errorResponse));
```

Comment:
This is actually a hack I tried following this https://github.com/github/fetch/issues/201#issuecomment-308213104 solution but then in catch clause(see down) I was receiving Promise which in turn contains the response object, and I wanted the response directly in catch clause.